src/ConnectedMode/UI/ManageConnections/ManageConnectionsViewModel.cs

```diff
@@ -56,17 +69,27 @@ internal bool InitializeConnections()
         {
             ConnectionViewModels.Clear();
             var succeeded = connectedModeServices.ServerConnectionsRepositoryAdapter.TryGetAllConnections(out var connections);
-            connections?.ForEach(AddConnection);
+            connections?.ForEach(AddConnectionViewModel);
             return succeeded;
         }
 
-        public void RemoveConnection(ConnectionViewModel connectionViewModel)
+        internal void RemoveConnection(ConnectionViewModel connectionViewModel)
         {
             ConnectionViewModels.Remove(connectionViewModel);
             RaisePropertyChanged(nameof(NoConnectionExists));
         }
 
-        public void AddConnection(Connection connection)
+        internal bool CreateNewConnection(Connection connection, ICredentialsModel credentialsModel)
+        {
+            var succeeded = connectedModeServices.ServerConnectionsRepositoryAdapter.TryAddConnection(connection, credentialsModel);
+            if (succeeded)
+            {
+                AddConnectionViewModel(connection);
+            }
+            return succeeded;
+        }
+
+        internal void AddConnectionViewModel(Connection connection)
```

Comment:
I think AddConnection is very confusing. This method really just creates the ViewModel and adds it to the list. It is only about the ViewModel, which updates the UI. I would not rename the method

Reply:
To me, it's the opposite, I find it confusing with the `ViewModel` suffix because we are already inside a `ViewModel`. By the context of the class all actions refer to the view.

Also, there is the `RemoveConnection` without the suffix, so it creates an inconsistency. If you have a strong opinion about that then it should be consistent across the class (rename the RemoveConnection and the InitializeConnections as well.)

In my opinion, the suffix is redundant.

Reply:
We have so many types of connection types (e.g. Connection, ConnectionInfo, ServerConnection) that I think being specific brings more clarity about what does the method do. I will update the RemoveConnection for consistency reasons.